app/Models/AlertSchedule.php

```diff
@@ -167,31 +168,38 @@ public function scopeIsActive($query)
+            $time_now = [$now, $local_now];
             $query->where('start', '<=', $now)
                 ->where('end', '>=', $now)
-                ->where(function ($query) use ($now) {
+                ->where(function ($query) use ($time_now) {
                     $query->where('recurring', 0) // Non recurring simply between start and end
-                    ->orWhere(function ($query) use ($now) {
+                    ->orWhere(function ($query) use ($time_now) {
                         $query->where('recurring', 1)
                             // Check the time is after the start date and before the end date, or end date is not set
-                            ->where(function ($query) use ($now) {
-                                $query->where(function ($query) use ($now) {
+                            ->where(function ($query) use ($time_now) {
+                                $query->where(function ($query) use ($time_now) {
                                     // normal, inside one day
                                     $query->whereTime('start', '<', DB::raw("time(`end`)"))
-                                        ->whereTime('start', '<=', $now->toTimeString())
-                                        ->whereTime('end', '>', $now->toTimeString());
-                                })->orWhere(function ($query) use ($now) {
+                                        ->whereTime('start', '<=', $time_now[0]->toTimeString())
+                                        ->whereTime('end', '>', $time_now[0]->toTimeString());
+                                })->orWhere(function ($query) use ($time_now) {
                                     // outside, spans days
                                     $query->whereTime('start', '>', DB::raw("time(`end`)"))
-                                        ->where(function ($query) use ($now) {
-                                            $query->whereTime('end', '<=', $now->toTimeString())
-                                                ->orWhereTime('start', '>', $now->toTimeString());
+                                        ->where(function ($query) use ($time_now) {
+                                            $query->where(function ($query) use ($time_now) {
+                                                $query->whereTime('start', '<=', $time_now[0]->toTimeString())
+                                                    ->whereTime(DB::raw("time(time(`end`) + 240000)"), '>', $time_now[0]->toTimeString());
+                                            })->orWhere(function ($query) use ($time_now) {
+                                                $query->whereTime('start', '<=', DB::raw("time(time(\"" . $time_now[0]->toTimeString(). "\") + 240000)"))
+                                                    ->whereTime(DB::raw("time(time(`end`) + 240000)"), '>', DB::raw("time(time(\"" . $time_now[0]->toTimeString() . "\") + 240000)"));
```

Comment:
Just use ->where() here, ->whereTime() causes an extra time() call since you are already calling time() in your raw query